ToolChain/Assembler/Parser/Parser.py

```diff
@@ -20,586 +19,369 @@
-                insParam = getattr(partialInstruction, param)
-                if insParam is not None:
-                    raise ValueError("Invalid instruction provided")
-
-        return instructionCode, instructionForm
-
-    def _generateBinaryInstructionFromInstruction(self, instruction: Instruction=None, form: dict=None):
+    def _evaluateFormBasedOnArguments(self, line, form, arglist):
         """
-        This is in charge of generating the actual binary code for an instruction. It will take each parts
-        of the instruction and build a binary number with it by using AND logic and bit shifts
-        :param instruction: Instruction, the instruction to work with
-        :param form: dict, a form dict that allows creation of the code
-        :return:
+        Method looks at the whole line after the initial instruction and determines its form based on the arguments.
+        Registers are concatenated as "REG", immediates and lables as "IMM", etc. We also populate a list of arguments
+        as they are to be constructed into binary code later.
+        :param line: str list, Line split into individual arguments
+        :param form: str, will be built up based on arguments on the line in this method
+        :param arglist: str list, Will contain our list of arguments to be used later when assembling our binary code
+        :return: Fully constructed form and newly populated list of arguments
         """
 
-        binaryInstruction = 0
-        instructionPartOriginal = None
-        instructionPartToBeUsed = None
-        memoryRefSymbol = None
-
-        for instructionPart in form["description"]:
-            shiftValue = 0x00  # This will be used in the last shifting operation for this part
-            instructionPartBitMask = form["description"][instructionPart]
-
-            # This is so we are able to use an and followed by a right shift to build the instruction
-            # easiest way that my brain could come up with. All other ways of building instructions
-            # seems to rely on ugly logic based on if chaining... Ugly. This is better. We are shifting
-            # until we find the "edge" of this part of the instruction.
-
-            while instructionPartBitMask % 2 is not 1:
-                instructionPartBitMask >>= 1  # Shift right one bit
-                shiftValue += 1
-
-            # We grab the value of the instruction to determine whether it's an actual instruction or a memory reference
-            instructionPartOriginal = getattr(instruction, instructionPart)
-            # We need an intermediate step because a part could be text
-
-            if type(instructionPartOriginal) is str:
-                # This is a memory reference
-                memoryRefSymbol = instructionPartOriginal
-                instructionPartToBeUsed = 0xDEADBEEF
+        argcount = 0
+
+        for arg in line[1:]:
+            if arg[0] == "$":
```

Comment:
Many elements in the next part of code should be defined in the assembler configuration file.